tests/kola/version/rhel-major-version

```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+# kola: { "exclusive": false }
+set -xeuo pipefail
+
+fatal() {
+    echo "$@" >&2
+    exit 1
+}
+
+#checks to ensure that the only packages from the RHEL major version are included
+source /etc/os-release
+
+if [ rpm -qa --queryformat='%{VERSION}\n' systemd ]; then
+    fatal "Error RHEL packages do not match current version"
+fi
```

Comment:
I don't think this is doing what you think it is doing.  It's mixing up a two query operations: `-qa` is used to fetch *all* the RPMs on a system and providing `systemd` you are asking for just `systemd`

This `if` statement will always succeed (i.e. return `0`) if `systemd` is installed in the OS...which unless RHCOS/RHEL changes drastically in the near future, will always be the case.

You need to set a variable to the RHEL major version, probably derived from the `RHEL_VERSION` in `/etc/os-release`.  (In this case the RHEL major version should be `8`)

Then you will want to query all the RPMs on the system to make sure they are for that RHEL major version.

Start by looking at `rpm -qa --queryformat='%{RELEASE}\n'` as an example of how to get the release value of all the installed packages.  Then sort out a way to compare the output to ensure that all the packages are targeting the correct RHEL major version.